File: kotlinlib/src/main/java/com/example/kotlinlib/bsase/java/NetManager.java
package com.example.kotlinlib.bsase.java;

// TODO　Java 单例模式1
public class NetManager {

    static class Holder {

        private static NetManager instance = new NetManager();

    }

    public static NetManager getInstance() {
        return Holder.instance;
    }

    public void show(String name) {
        System.out.println("show:" + name);
    }





    // 使用端使用
    public static void client() {
       NetManager.getInstance().show("Derry1");
    }
    public static void main(String[] args) {
        client();
    }

}

File: kotlinlib/src/main/java/com/example/kotlinlib/bsase/simple05/s04/NetManager.kt
package simple05.s04
//模仿静态内部类写的；
// TODO　Kotlin 单例模式1
class NetManager {

    // 只有一个实例
    object Holder {

        val instance = NetManager()

    }

    // 看不到 static  可以 派生操作
    companion object {

        // 全部都是  相当于 Java static

        fun getInstance() : NetManager = Holder.instance
    }

    fun show(name: String) {
        println("show:$name");
    }

}

// 客户端
fun main() {
    val net  = NetManager.getInstance()
    net.show("kt Derry1")
}